client/main.lua

```diff
@@ -433,6 +433,7 @@ local function init()
     registerEnterRoof()
     registerExitRoof()
 
+	if not isLoggedIn then return end
```

Comment:
So rather than blocking the zone creation if you don't match the job we should put the job check inside these respective zones. The onEnter and inside functions can have the `if QBX.PlayerData.job.name ~= 'reporter' then return end` added to them to early return.